spdm.rflx

```diff
@@ -0,0 +1,1487 @@
+               if Registry_ID = Reg_IANA
+                  or Registry_ID = Reg_HD_Base_T;
+         Algorithm_ID_16 : Algorithm_ID_16
+            then null;
+         Algorithm_ID_32 : Algorithm_ID_32;
+      end message;
+
+   type Ext_Algs is sequence of Ext_Alg;
+
+   -- [187] Algorithm request structure
+
+   type Alg_Type is
+      (DHE               => 2,
+       AEAD_Cipher_Suite => 3,
+       Req_Base_Asym_Alg => 4,
+       Key_Schedule      => 5)
+   with Size => 8;
+
+   type Alg_Count is range 0 .. 2**4 - 1 with Size => 4;
+
+   type Alg_Struct is
+      message
+         Alg_Type : Alg_Type;
+         Fixed_Alg_Count : Alg_Count
+            then Ext_Alg_Count
+               if Fixed_Alg_Count = 2;
+         Ext_Alg_Count : Alg_Count
+            then DHE_Reserved_1
+               if Alg_Type = DHE
+            then AEAD_Reserved_1
+               if Alg_Type = AEAD_Cipher_Suite
+            then RBAA_TPM_ALG_ECDSA_ECC_NIST_P384
+               if Alg_Type = Req_Base_Asym_Alg
+            then KS_Reserved_1
+               if Alg_Type = Key_Schedule;
+         DHE_Reserved_1 : Reserved_2;
+         DHE_SecP521r1 : Boolean;
+         DHE_SecP384r1 : Boolean;
+         DHE_SecP256r1 : Boolean;
+         DHE_FFDHE4096 : Boolean;
+         DHE_FFDHE3072 : Boolean;
+         DHE_FFDHE2048 : Boolean;
+         DHE_Reserved_2 : Reserved_8
+            then Alg_External;
+         AEAD_Reserved_1 : Reserved_5;
+         AEAD_ChaCha20_Poly1305 : Boolean;
+         AEAD_AES_256_GCM : Boolean;
+         AEAD_AES_128_GCM : Boolean;
+         AEAD_Reserved_2 : Reserved_8
+            then Alg_External;
+         RBAA_TPM_ALG_ECDSA_ECC_NIST_P384 : Boolean;
+         RBAA_TPM_ALG_RSAPSS_4096 : Boolean;
+         RBAA_TPM_ALG_RSASSA_4096 : Boolean;
+         RBAA_TPM_ALG_ECDSA_ECC_NIST_P256 : Boolean;
+         RBAA_TPM_ALG_RSAPSS_3072 : Boolean;
+         RBAA_TPM_ALG_RSASSA_3072 : Boolean;
+         RBAA_TPM_ALG_RSAPSS_2048 : Boolean;
+         RBAA_TPM_ALG_RSASSA_2048 : Boolean;
+         RBAA_Reserved_1 : Reserved_7;
+         RBAA_TPM_ALG_ECDSA_ECC_NIST_P521 : Boolean
+            then Alg_External;
+         KS_Reserved_1 : Reserved_7;
+         KS_SPDM : Boolean;
+         KS_Reserved_2 : Reserved_8
+            then Alg_External;
+         Alg_External : Ext_Algs
+            with Size => 32 * Ext_Alg_Count;
+      end message;
+
+   type Req_Alg_Structs is sequence of Alg_Struct;
+
+   -- 10.4 NEGOTIATE_ALGORITHMS request and ALGORITHMS response messages
+
+   type Alg_Struct_Count is range 0 .. 2**8 - 1 with Size => 8;
+   type Ext_Asym_Count is range 0 .. 2**8 - 1 with Size => 8;
+   type Ext_Hash_Count is range 0 .. 2**8 - 1 with Size => 8;
+
+   --  ISSUE: Componolit/RecordFlux#609
+   --  This message differs between SPDM version 1.0 and 1.1.
+   --  Since the minor version is not accessible from
+   --  within this message one message is needed for each version
+   --  that is selected by a refinement.
```

Comment:
Componolit/RecordFlux#609 is fixed. This actually is part of #92 where the duplicate message should be merged again.

Reply:
Fixed in https://github.com/Componolit/RecordFlux-specifications/pull/93/commits/7c1908a0db7eece9cbbb987fa14b0c9d9e4ec91f.